matter_server/server/client_handler.py

```diff
@@ -182,3 +133,84 @@ async def async_handle(self) -> web.WebSocketResponse:
 
         return wsock
+
+    def _handle_command(self, msg: CommandMessage) -> None:
+        """Handle an incoming command from the client."""
+        self._logger.debug("Handling command %s", msg.command)
+
+        # work out handler for the given path/command
+        handler = self.server.command_handlers.get(msg.command)
+
+        if handler is None:
+            self._send_message(
+                ErrorResultMessage(
+                    msg.messageId,
+                    ErrorCode.INVALID_COMMAND,
+                    f"Invalid command: {msg.command}",
+                )
+            )
+            self._logger.warning("Invalid command: %s", msg.command)
+            return
+
+        # schedule task to handle the command
+        self.server.loop.create_task(self._run_handler(handler, msg))
+
+    async def _run_handler(
+        self, handler: APICommandHandler, msg: CommandMessage
+    ) -> None:
+        try:
+            args = parse_arguments(handler.signature, msg.args)
+            result = handler.target(**args)
+            if asyncio.iscoroutine(result):
+                result = await result
+            self._send_message(SuccessResultMessage(msg.messageId, result))
+        except ChipStackError as err:
+            self._send_message(
+                ErrorResultMessage(msg.messageId, ErrorCode.STACK_ERROR, str(err))
+            )
+        except Exception as err:  # pylint: disable=broad-except
+            self._logger.exception("Error handling message: %s", msg)
+            self._send_message(
+                ErrorResultMessage(msg.messageId, ErrorCode.UNKNOWN_ERROR, str(err))
+            )
+
+    async def _writer(self) -> None:
+        """Write outgoing messages."""
+        # Exceptions if Socket disconnected or cancelled by connection handler
+        with suppress(RuntimeError, ConnectionResetError, *CANCELLATION_ERRORS):
+            while not self.wsock.closed:
+                if (process := await self._to_write.get()) is None:
+                    break
+
+                if not isinstance(process, str):
+                    message: str = process()
+                else:
+                    message = process
+                self._logger.debug("Sending %s", message)
+                await self.wsock.send_str(message)
+
+    def _send_message(self, message: MessageType) -> None:
+        """
+        Send a message to the client.
+
+        Closes connection if the client is not reading the messages.
+
+        Async friendly.
+        """
+        message = json_dumps(message)
+
+        try:
+            self._to_write.put_nowait(message)
+        except asyncio.QueueFull:
+            self._logger.error(
+                "Client exceeded max pending messages: %s", MAX_PENDING_MSG
+            )
+
+            self._cancel()
+
+    def _cancel(self) -> None:
+        """Cancel the connection."""
+        if self._handle_task is not None:
+            self._handle_task.cancel()
```

Comment:
Do we need to await the task too to make sure it's cancelled, or is that not needed?

Reply:
nah, in here it is not that important I guess